cgvclone/src/main/java/com/ceos22/spring_boot/common/auth/security/authentication/SaltAuthenticationProvider.java

```diff
@@ -0,0 +1,51 @@
+package com.ceos22.spring_boot.common.auth.security.authentication;
+
+import com.ceos22.spring_boot.common.auth.security.principal.CustomUserPrincipal;
+import com.ceos22.spring_boot.common.auth.security.password.PasswordSaltService;
+import com.ceos22.spring_boot.common.exception.AuthFailureException;
+import com.ceos22.spring_boot.common.response.status.ErrorStatus;
+import com.ceos22.spring_boot.domain.user.User;
+import com.ceos22.spring_boot.domain.user.UserRepository;
+import org.springframework.security.authentication.*;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class SaltAuthenticationProvider implements AuthenticationProvider {
+    private final UserRepository users;
+    private final PasswordSaltService saltService;
+
+    public SaltAuthenticationProvider(UserRepository users, PasswordSaltService saltService) {
+        this.users = users; this.saltService = saltService;
+    }
+
+    @Override
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+        String username = String.valueOf(authentication.getPrincipal());
+        String rawPassword = String.valueOf(authentication.getCredentials());
+
+        User u = users.findByUsername(username)
```

Comment:
p2. 축약어 사용은 지양하면 좋을 것 같아요.
코드가 길어지거나 여러 축약어르 사용하게 되면, 단어가 잘 떠오르지 않게 되어 가독성이 떨어집니다

```suggestion
        User user = users.findByUsername(username)
```